src/theme/MDXComponents.js

```
@@ -1,70 +1,67 @@
+import GeekDetails from '@site/src/components/partials/GeekDetails';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import { useLocation } from '@docusaurus/router';
 
-const partials = {
-  ReqsSys,
-  ReqsKernel,
-  HostEndpointsUpgrade,
-  InstallOpenshiftBeforeYouBegin,
-  CalicoWindowsInstall,
-  PodCidrSed,
-  EnvironmentFile,
-  AutoHostendpointsMigrate,
-  ConfigureManagedCluster,
-  InstallAKS,
-  InstallEKS,
-  InstallGeneric,
-  InstallGKE,
-  PrivateRegistryRegular,
-  UpgradeOperatorSimple,
-  InstallOpenShift,
-  InstallOpenShiftManifests,
-  OpenShiftPullSecret,
-  OpenShiftPrometheusOperator,
-  GeekDetails,
-  ReleaseNotesCalico,
-  ReleaseNotesCalicoEnterprise,
-  CliConfigIntro,
+const wrap = (Partial) => (props) => {
+  return (
+    <div className='docs-partial'>
+      <Partial {...props} />
+    </div>
+  );
 };
 
-const wrappedPartials = wrapPartials(partials);
+const resolveComponent = (componentName) => {
+  return (props) => {
+    const { pathname } = useLocation();
+    const context = useDocusaurusContext();
 
-export default {
-  ...MDXComponents,
-  ...wrappedPartials,
+    const [, prodnamedash, maybeVersion] = pathname.match(/\/(.*?)\/(.*?)\//);
+    const [, ...versions] = context.globalData['docusaurus-plugin-content-docs'][prodnamedash].versions;
+
+    let Component = () => '';
+    const isNext = maybeVersion === 'next';
+    const noVersions = !versions || !versions.length;
+
+    if (isNext || noVersions) {
+      Component = require(`@site/${prodnamedash}/_includes/components/${componentName}`).default;
+    } else {
+      const isLatest = !versions.some((v) => v.name === maybeVersion);
+      const version = isLatest ? versions.find((v) => v.isLast).name : maybeVersion;
+      Component =
+        require(`@site/${prodnamedash}_versioned_docs/version-${version}/_includes/components/${componentName}`).default;
+    }
+
+    return <Component {...props} />;
```

Please throw an error for safety reasons if there is no resolved component.